ai21/logger.py

```diff
@@ -1,8 +1,64 @@
 import logging
+import os
+import re
 
 from ai21.ai21_env_config import AI21EnvConfig
 
+_verbose = False
+
 logger = logging.getLogger("ai21")
+httpx_logger = logging.getLogger("httpx")
+
+
+class CensorSecretsFormatter(logging.Formatter):
+    def format(self, record: logging.LogRecord) -> str:
+        # Get original log message
+        message = super().format(record)
+
+        if not get_verbose():
+            return self._censor_secrets(message)
+
+        return message
+
+    def _censor_secrets(self, message: str) -> str:
+        # Regular expression to find the Authorization key and its value
+        pattern = r"('Authorization':\s*'[^']*'|'api-key':\s*'[^']*')"
```

Comment:
Since we're logging headers, we'd need to also add to the pattern `X-Amz-Security-Token` as this is one of the auth headers for aws